zerver/tests/test_notification_data.py

```diff
@@ -113,6 +113,23 @@ def test_is_push_notifiable(self) -> None:
         )
         self.assertFalse(user_data.is_push_notifiable(acting_user_id=acting_user_id, idle=True))
 
+        # Message sender is the welcome bot.
```

Comment:
This comment needs updating -- it should explain something like "disable_external_notifications takes precedence over  other flags."